dataset/src/test/scala/frameless/functions/UdfTest.scala

```diff
@@ -0,0 +1,194 @@
+package frameless
+package functions
+
+import org.scalacheck.Prop
+import org.scalacheck.Prop._
+
+class UdfTest extends TypedDatasetSuite {
+
+  test("one argument udf") {
+    def prop[A, B, C](data: Vector[X3[A, B, C]], f1: A => A)(
```

Comment:
Good catch, I can make it `def prop[A: TypedEncoder, B: TypedEncoder, C: TypedEncoder]`. You think that doing so it will automatically extract an encoding for X3[A,B,C]? Do I need to include something here? I think I tried it and it was complaining.